> Makefiles are black magic, it is well known in the industry, is there an easy way to solve these warnings?
> 
> ```
> simson@tama:~/polkadex/polkadexTEE-worker$ make benchmark
> Makefile:208: warning: overriding recipe for target 'enclave/Enclave_t.c'
> Makefile:154: warning: ignoring old recipe for target 'enclave/Enclave_t.c'
> Makefile:208: warning: overriding recipe for target 'enclave/Enclave_t.h'
> Makefile:154: warning: ignoring old recipe for target 'enclave/Enclave_t.h'
> Makefile:208: warning: overriding recipe for target 'worker/Enclave_u.c'
> Makefile:154: warning: ignoring old recipe for target 'worker/Enclave_u.c'
> Makefile:208: warning: overriding recipe for target 'worker/Enclave_u.h'
> Makefile:154: warning: ignoring old recipe for target 'worker/Enclave_u.h'
> Makefile:214: warning: overriding recipe for target 'worker/Enclave_u.o'
> Makefile:160: warning: ignoring old recipe for target 'worker/Enclave_u.o'
> Makefile:218: warning: overriding recipe for target 'worker/libEnclave_u.a'
> Makefile:164: warning: ignoring old recipe for target 'worker/libEnclave_u.a'
> Makefile:222: warning: overriding recipe for target 'bin/app'
> Makefile:168: warning: ignoring old recipe for target 'bin/app'
> Makefile:231: warning: overriding recipe for target 'bin/substratee-client'
> Makefile:177: warning: ignoring old recipe for target 'bin/substratee-client'
> Makefile:239: warning: overriding recipe for target 'enclave/Enclave_t.o'
> Makefile:185: warning: ignoring old recipe for target 'enclave/Enclave_t.o'
> Makefile:243: warning: overriding recipe for target 'enclave/enclave.so'
> Makefile:189: warning: ignoring old recipe for target 'enclave/enclave.so'
> Makefile:248: warning: overriding recipe for target 'bin/enclave.signed.so'
> Makefile:194: warning: ignoring old recipe for target 'bin/enclave.signed.so'
> Makefile:266: warning: overriding recipe for target 'enclave'
> Makefile:257: warning: ignoring old recipe for target 'enclave'
> Makefile:271: warning: overriding recipe for target '.git/hooks/pre-commit'
> Makefile:262: warning: ignoring old recipe for target '.git/hooks/pre-commit'
> ```

Haha .. Yes @simonsso. Hey @haerdib could you please help me with these warnings. It is first time, I am working with it.😅

--- a/Makefile
+++ b/Makefile
@@ -22,6 +22,7 @@ SGX_ARCH ?= x64
 SGX_DEBUG ?= 0
 SGX_PRERELEASE ?= 0
 SGX_PRODUCTION ?= 0
+BENCHMARK ?= 0
 
 SKIP_WASM_BUILD = 1
 # include the build settings from rust-sgx-sdk
@@ -197,70 +198,7 @@ $(Signed_RustEnclave_Name): $(RustEnclave_Name)
 	@echo "SIGN =>  $@"
 	@echo
 	@echo "Enclave is in $(SGX_ENCLAVE_MODE)"
-.PHONY: benchmark
-benchmark: $(Client_Name) $(Worker_Name) $(Signed_RustEnclave_Name)
-worker: $(Worker_Name)
-client: $(Client_Name)
-githooks: .git/hooks/pre-commit
 
-######## EDL objects ########
-$(Enclave_EDL_Files): $(SGX_EDGER8R) enclave/Enclave.edl
-	$(SGX_EDGER8R) --trusted enclave/Enclave.edl --search-path $(SGX_SDK)/include --search-path $(CUSTOM_EDL_PATH) --trusted-dir enclave
-	$(SGX_EDGER8R) --untrusted enclave/Enclave.edl --search-path $(SGX_SDK)/include --search-path $(CUSTOM_EDL_PATH) --untrusted-dir worker
-	@echo "GEN  =>  $(Enclave_EDL_Files)"
-
-######## SubstraTEE-worker objects ########
-worker/Enclave_u.o: $(Enclave_EDL_Files)
-	@$(CC) $(Worker_C_Flags) -c worker/Enclave_u.c -o $@
-	@echo "CC   <=  $<"
-
-$(Worker_Enclave_u_Object): worker/Enclave_u.o
-	$(AR) rcsD $@ $^
-	cp $(Worker_Enclave_u_Object) ./lib
-
-$(Worker_Name): $(Worker_Enclave_u_Object) $(Worker_SRC_Files)
-	@echo
-	@echo "Building the substraTEE-worker"
-	@cd worker && SGX_SDK=$(SGX_SDK) cargo build $(Worker_Rust_Flags)
-	@echo "Cargo  =>  $@"
-	cp $(Worker_Rust_Path)/substratee-worker ./bin
-	cp $(Worker_Rust_Path)/substratee-worker ./bin2
-
-######## SubstraTEE-client objects ########
-$(Client_Name): $(Client_SRC_Files)
-	@echo
-	@echo "Building the substraTEE-client"
-	@cd $(Client_SRC_Path) && cargo build $(Client_Rust_Flags)
-	@echo "Cargo  =>  $@"
-	cp $(Client_Rust_Path)/$(Client_Binary) ./bin
-
-######## Enclave objects ########
-enclave/Enclave_t.o: $(Enclave_EDL_Files)
-	@$(CC) $(RustEnclave_Compile_Flags) -c enclave/Enclave_t.c -o $@
-	@echo "CC   <=  $<"
-
-$(RustEnclave_Name): enclave-benchmark enclave/Enclave_t.o
-	@echo Compiling $(RustEnclave_Name)
-	@$(CXX) enclave/Enclave_t.o -o $@ $(RustEnclave_Link_Flags)
-	@echo "LINK =>  $@"
-
-$(Signed_RustEnclave_Name): $(RustEnclave_Name)
-	@echo
-	@echo "Signing the enclave: $(SGX_ENCLAVE_MODE)"
-	$(SGX_ENCLAVE_SIGNER) sign -key $(SGX_SIGN_KEY) -enclave $(RustEnclave_Name) -out $@ -config $(SGX_ENCLAVE_CONFIG)
-	@echo "SIGN =>  $@"
-	@echo
-	@echo "Enclave is in $(SGX_ENCLAVE_MODE)"
-
-.PHONY: enclave-benchmark
-enclave:
-	@echo
-	@echo "Building the enclave"
-	$(MAKE) -C ./enclave/ benchmark
-
-.git/hooks/pre-commit: .githooks/pre-commit
-	@echo "Installing git hooks"
-	cp .githooks/pre-commit .git/hooks
 .PHONY: enclave
 enclave:
 	@echo

--- a/enclave/Makefile
+++ b/enclave/Makefile
@@ -38,6 +38,10 @@ else
 	CARGO_TARGET := --release
 endif
 
+ifeq ($(BENCHMARK), 1)
+		CARGO_TARGET += --features "benchmarking"
+endif
+
 .PHONY: all
 
 all: $(Rust_Enclave_Name)
@@ -48,18 +52,5 @@ ifeq ($(XARGO_SGX), 1)
 	cp ./target/x86_64-unknown-linux-sgx/$(OUTPUT_PATH)/libsubstratee_worker_enclave.a ../lib/libenclave.a
 else
 	cargo build $(CARGO_TARGET)
-	cp ./target/$(OUTPUT_PATH)/libsubstratee_worker_enclave.a ../lib/libenclave.a
-endif
-
-.PHONY: benchmark
-
-benchmark: $(Rust_Enclave_Name)
-
-$(Rust_Enclave_Name): $(Rust_Enclave_Files)
-ifeq ($(XARGO_SGX), 1)
-	RUST_TARGET_PATH=$(Rust_Target_Path) xargo build --target x86_64-unknown-linux-sgx $(CARGO_TARGET)
-	cp ./target/x86_64-unknown-linux-sgx/$(OUTPUT_PATH)/libsubstratee_worker_enclave.a ../lib/libenclave.a
-else
-	cargo build $(CARGO_TARGET) --features "benchmarking"
 	cp ./target/$(OUTPUT_PATH)/libsubstratee_worker_enclave.a ../lib/libenclave.a
 endif

